app/reflexes/products_reflex.rb

```diff
@@ -3,36 +3,52 @@
 class ProductsReflex < ApplicationReflex
   include Pagy::Backend
 
-  def fetch
-    @page ||= element.dataset.page || 1
-    @per_page ||= element.dataset.perpage || 15
-    @search_term ||= element.dataset.searchterm || ""
-
-    fetch_products
+  before_reflex :init_params
 
-    render_products
+  def fetch
+    fetch_and_render_products
   end
 
   def change_per_page
     @per_page = element.value.to_i
     @page = 1
 
-    fetch
+    fetch_and_render_products
   end
 
   def filter
-    @per_page = params[:per_page]
     @page = 1
-    @search_term = params[:search_term]
-    @producer_id = params[:producer_id]
-    @category_id = params[:category_id]
 
-    fetch_products
-    render_products
+    fetch_and_render_products
   end
 
   private
 
+  def init_params
+    init_filters_params
+    init_pagination_params
+  end
+
+  def init_filters_params
+    # params comes from the form
+    # _params comes from the url
+    # priority is given to params from the form (if present) over url params
+    @search_term = params[:search_term] || params[:_search_term]
+    @producer_id = params[:producer_id] || params[:_producer_id]
+    @category_id = params[:category_id] || params[:_category_id]
```

Comment:
It's a shame we need to manage these this way.. surely there is a better way? Can the JS attached to the "search" button somehow submit only the form params in the request?

Reply:
I can't make it work this way either. Would you like to give a try? Maybe this needs a fresh brain

Reply:
I don't think I have a fresh brain to offer :P
I did a little bit of googling and couldn't find an answer.